### ISSUE TITLE
Absolute filePaths used in output

### ISSUE THREAD

**Issue by kimburgess** [textlint/textlint#769]
**What version of textlint are you using?**
`11.9.1`

When running textlint, outputted results list absolute paths. Potential cause: https://github.com/textlint/textlint/blob/538b64e2ebadca8e027972233f2bf82c0188570a/packages/textlint/src/util/find-util.ts#L78-L87

This behaviour deviates from the common linting tools and breaks compatibility with some commonly useful tooling, such as default configurations for [GitHub problem matchers](https://github.com/actions/toolkit/blob/master/docs/problem-matchers.md).

If absolute paths are desirable in some use cases, providing a CLI option to toggle between this and their relative equivalents would be beneficial to reducing external tooling requirements.

**Comment by azu** [comment]
If the usecase is only GitHub problem matchers, we may create new formatter like `eslint-stylish` in https://github.com/textlint/textlint/tree/master/packages/%40textlint/linter-formatter.

The current `stylish` format is a little different from eslint's `stylish`.
(textlint add `✔`  to the resulting output when it is fixable. Related blog <https://qiita.com/nanakenashi/items/1797bedfa343833173f2>)

**Comment by azu** [comment]
Of course, It is good that we think about relative path outputs options.

Related ESLint Issue

- [Output relative paths rather than absolute paths · Issue #13376 · eslint/eslint](https://github.com/eslint/eslint/issues/13376)
  - [Accessing `cwd` from formatters by mysticatea · Pull Request #57 · eslint/rfcs](https://github.com/eslint/rfcs/pull/57)
  - [Update: pass cwd to formatters (refs eslint/rfcs#57) by mysticatea · Pull Request #13392 · eslint/eslint](https://github.com/eslint/eslint/pull/13392)

**Comment by kimburgess** [comment]
While the `stylish` formatter has the incompatibility you note, `compact` does look to play nicely with [`eslint-compact`](https://github.com/actions/setup-node/blob/main/.github/eslint-compact.json).

It looks like the absolute paths _may_ be compatible with the default matchers available in Actions, but have potentially hit another issue (https://github.com/actions/toolkit/issues/810), unrelated to Textlint.

There are other cases where relative paths are useful so supporting that would be a wonderful feature.

**Comment by azu** [comment]
> There are other cases where relative paths are useful so supporting that would be a wonderful feature.

Yes, It is the best result.
However, textlint has `--output-file` flag. `--output-file` option may make it difficult to output relative file path.

Probably, we need to pass some missing info like that is output-file path and `cwd` to formatted.

:memo: [tsuyoshicho/action-textlint](https://github.com/tsuyoshicho/action-textlint) provide another way for review comments on GitHub.
